tasks/terraform/src/main/java/com/walmartlabs/concord/plugins/terraform/Terraform.java

```diff
@@ -42,34 +44,66 @@ public class Terraform {
     private static final Logger log = LoggerFactory.getLogger(Terraform.class);
 
     private final boolean debug;
+    private final Path processWorkDir;
     private final Path binary;
     private final Map<String, String> baseEnv;
     private final ExecutorService executor;
     private final Version version;
+    private final String dockerImage;
+    private final TerraformDockerService dockerService;
+
+    public enum CLI_ACTION {
+        APPLY,
+        DESTROY,
+        INIT,
+        OUTPUT,
+        PLAN,
+        VERSION
+    }
 
     /**
      * @param debug   enable/disable additional debug output
      */
-    public Terraform(TerraformBinaryResolver binaryResolver, boolean debug, Map<String, String> baseEnv) throws Exception {
+    public Terraform(Path processWorkDir,
+                     TerraformBinaryResolver binaryResolver,
+                     boolean debug,
+                     Map<String, String> baseEnv,
+                     String dockerImage,
+                     TerraformDockerService dockerService) throws Exception {
         this.debug = debug;
+        this.processWorkDir = processWorkDir;
         this.baseEnv = baseEnv;
         this.binary = binaryResolver.resolve();
         this.executor = Executors.newCachedThreadPool();
+        this.dockerImage = dockerImage;
+        this.dockerService = dockerService;
         this.version = getBinaryVersion();
+
+        if (dockerImage != null) {
+            log.info("Executing terraform commands in container using image: {}", dockerImage);
```

Comment:
Might make more sense to move this log statement to execDocker method maybe?

Reply:
I have a slight preference for this location for cutting down on non-debug noise. If it's in `dockerExec()`, it'll get printed multiple times for each task call since we call `terraform ...` multiple times. If `debug: true`, you'll get tons of docker indicators all over the place.
```12:16:16 [INFO ] exec -> /workspace/.terraform/terraform version in hub.docker.prod.walmart.com/zenika/terraform-aws-cli:latest/tmp/concord-agent/workDirs/71efde85-b17d-4b68-b580-9616f5bcda59
12:16:16 [INFO ] exec -> using env: {TF_PLUGIN_CACHE_DIR=/workspace/.terraform/plugin-cache, GIT_SSH_COMMAND=/tmp/concord-agent/workDirs/71efde85-b17d-4b68-b580-9616f5bcda59/gitSsh4763013649172513336.sh}
12:16:16 [INFO ] CMD: [/bin/sh, -c, docker run -u 456 --rm -i -v '/tmp/concord-agent/workDirs/71efde85-b17d-4b68-b580-9616f5bcda59:/workspace' -e 'DOCKER_HOST=tcp://concord-dind:6666' -e 'TF_PLUGIN_CACHE_DIR=/workspace/.terraform/plugin-cache' -e 'GIT_SSH_COMMAND=/tmp/concord-agent/workDirs/71efde85-b17d-4b68-b580-9616f5bcda59/gitSsh4763013649172513336.sh' -w '/workspace' --label 'concordTxId=71efde85-b17d-4b68-b580-9616f5bcda59' -v /tmp/passwd5532242361710637800.docker:/etc/passwd:ro --net=host 'hub.docker.prod.walmart.com/zenika/terraform-aws-cli:latest' '/workspace/.terraform/terraform' 'version']
```